src/main/scala/com.snowplowanalytics.snowplow.analytics.scalasdk/json/EventTransformer.scala

```diff
@@ -94,138 +166,273 @@ object EventTransformer {
-    "mkt_content" -> StringField,
-    "mkt_campaign" -> StringField,
-    "contexts" -> CustomContextsField,
-    "se_category" -> StringField,
-    "se_action" -> StringField,
-    "se_label" -> StringField,
-    "se_property" -> StringField,
-    "se_value" -> DoubleField,
-    "unstruct_event" -> UnstructField,
-    "tr_orderid" -> StringField,
-    "tr_affiliation" -> StringField,
-    "tr_total" -> DoubleField,
-    "tr_tax" -> DoubleField,
-    "tr_shipping" -> DoubleField,
-    "tr_city" -> StringField,
-    "tr_state" -> StringField,
-    "tr_country" -> StringField,
-    "ti_orderid" -> StringField,
-    "ti_sku" -> StringField,
-    "ti_name" -> StringField,
-    "ti_category" -> StringField,
-    "ti_price" -> DoubleField,
-    "ti_quantity" -> IntField,
-    "pp_xoffset_min" -> IntField,
-    "pp_xoffset_max" -> IntField,
-    "pp_yoffset_min" -> IntField,
-    "pp_yoffset_max" -> IntField,
-    "useragent" -> StringField,
-    "br_name" -> StringField,
-    "br_family" -> StringField,
-    "br_version" -> StringField,
-    "br_type" -> StringField,
-    "br_renderengine" -> StringField,
-    "br_lang" -> StringField,
-    "br_features_pdf" -> BoolField,
-    "br_features_flash" -> BoolField,
-    "br_features_java" -> BoolField,
-    "br_features_director" -> BoolField,
-    "br_features_quicktime" -> BoolField,
-    "br_features_realplayer" -> BoolField,
-    "br_features_windowsmedia" -> BoolField,
-    "br_features_gears" -> BoolField,
-    "br_features_silverlight" -> BoolField,
-    "br_cookies" -> BoolField,
-    "br_colordepth" -> StringField,
-    "br_viewwidth" -> IntField,
-    "br_viewheight" -> IntField,
-    "os_name" -> StringField,
-    "os_family" -> StringField,
-    "os_manufacturer" -> StringField,
-    "os_timezone" -> StringField,
-    "dvce_type" -> StringField,
-    "dvce_ismobile" -> BoolField,
-    "dvce_screenwidth" -> IntField,
-    "dvce_screenheight" -> IntField,
-    "doc_charset" -> StringField,
-    "doc_width" -> IntField,
-    "doc_height" -> IntField,
-    "tr_currency" -> StringField,
-    "tr_total_base" -> DoubleField,
-    "tr_tax_base" -> DoubleField,
-    "tr_shipping_base" -> DoubleField,
-    "ti_currency" -> StringField,
-    "ti_price_base" -> DoubleField,
-    "base_currency" -> StringField,
-    "geo_timezone" -> StringField,
-    "mkt_clickid" -> StringField,
-    "mkt_network" -> StringField,
-    "etl_tags" -> StringField,
-    "dvce_sent_tstamp" -> TstampField,
-    "refr_domain_userid" -> StringField,
-    "refr_device_tstamp" -> TstampField,
-    "derived_contexts" -> DerivedContextsField,
-    "domain_sessionid" -> StringField,
-    "derived_tstamp" -> TstampField,
-    "event_vendor" -> StringField,
-    "event_name" -> StringField,
-    "event_format" -> StringField,
-    "event_version" -> StringField,
-    "event_fingerprint" -> StringField,
-    "true_tstamp" -> TstampField
+  private lazy val Fields = FieldNames.zip(FieldTypes)
+
+  private val FieldNames = List(
```

Comment:
Having `name -> Type` was a nice reference. Wondering why it was necessary to split it, couldn't we go with `val FieldNames = Fields.map(_._1)`

Reply:
Yes. This is "early version" and this was top of my todo list to fix as it is no longer relevant